src/OpenTelemetry.Exporter.Geneva/Metrics/GenevaMetricExporter.cs

```diff
@@ -484,6 +486,384 @@ internal static void DisableOpenTelemetrySdkMetricNameValidation()
+        string metricName,
+        long timestamp,
+        in ReadOnlyTagCollection tags,
+        MetricData value,
+        Exemplar[] exemplars)
+    {
+        ushort bodyLength;
+        try
+        {
+            // The buffer format is as follows:
+            // -- BinaryHeader
+            // -- Sequence of payload types
+
+            // Leave enough space for the header
+            var bufferIndex = sizeof(BinaryHeader);
+
+            // Serialize metric name
+            MetricSerializer.SerializeByte(this.buffer, ref bufferIndex, (byte)PayloadType.MetricName);
+            MetricSerializer.SerializeEncodedString(this.buffer, ref bufferIndex, Encoding.UTF8.GetBytes(metricName));
+
+            #region Serialize metric data
+
+            var payloadType = eventType == MetricEventType.ULongMetric ? PayloadType.ULongMetric : PayloadType.DoubleMetric;
+            MetricSerializer.SerializeByte(this.buffer, ref bufferIndex, (byte)payloadType);
+
+            // Get a placeholder to add the payloadType length
+            int payloadTypeStartIndex = bufferIndex;
+            bufferIndex += 2;
+
+            MetricSerializer.SerializeUInt64(this.buffer, ref bufferIndex, (ulong)timestamp); // timestamp
+
+            if (payloadType == PayloadType.ULongMetric)
+            {
+                MetricSerializer.SerializeUInt64(this.buffer, ref bufferIndex, value.UInt64Value);
+            }
+            else
+            {
+                MetricSerializer.SerializeFloat64(this.buffer, ref bufferIndex, value.DoubleValue);
+            }
+
+            var payloadTypeLength = (ushort)(bufferIndex - payloadTypeStartIndex - 2);
+            MetricSerializer.SerializeUInt16(this.buffer, ref payloadTypeStartIndex, payloadTypeLength);
+
+            #endregion
+
+            #region Serialize metric dimensions
+            MetricSerializer.SerializeByte(this.buffer, ref bufferIndex, (byte)PayloadType.Dimensions);
+
+            // Get a placeholder to add the payloadType length
+            payloadTypeStartIndex = bufferIndex;
+            bufferIndex += 2;
+
+            // Get a placeholder to add dimensions count later
+            var bufferIndexForDimensionsCount = bufferIndex;
+            bufferIndex += 2;
+
+            ushort dimensionsWritten = 0;
+
+            // Serialize PrepopulatedDimensions keys
+            for (ushort i = 0; i < this.prepopulatedDimensionsCount; i++)
+            {
+                MetricSerializer.SerializeEncodedString(this.buffer, ref bufferIndex, this.serializedPrepopulatedDimensionsKeys[i]);
+            }
+
+            if (this.prepopulatedDimensionsCount > 0)
+            {
+                dimensionsWritten += this.prepopulatedDimensionsCount;
+            }
+
+            // Serialize MetricPoint Dimension keys
+            foreach (var tag in tags)
+            {
+                if (tag.Key.Length > MaxDimensionNameSize)
+                {
+                    // TODO: Data Validation
```

Comment:
this is same as existing behavior right? i.e e drop whole metric due to excess key?

Reply:
This follows the existing behavior but the existing behavior is not to drop the metric. We simply serialize the tag key to the buffer and if the whole payload length is still under 64KB, we export it. It's then up to the downstream systems to handle it.